src/uct/cuda/base/cuda_iface.h

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2018.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCT_CUDA_IFACE_H
+#define UCT_CUDA_IFACE_H
+
+#include <ucs/sys/preprocessor.h>
+#include <cuda_runtime.h>
+#include <cuda.h>
+
+#define UCT_CUDA_FUNC(_func)                                    \
+    ({                                                          \
+        ucs_status_t _status = UCS_OK;                          \
+        do {                                                    \
+            cudaError_t _result = (_func);                      \
+            if (cudaSuccess != _result) {                       \
+                ucs_error("%s is failed. ret:%s",               \
+                          UCS_PP_MAKE_STRING(_func),            \
+                          cudaGetErrorString(_result));         \
+                _status = UCS_ERR_IO_ERROR;                     \
+            }                                                   \
+        } while (0);                                            \
+        _status;                                                \
+    })
+
+
+#define UCT_CUDADRV_FUNC(_func)                                 \
```

Comment:
Can we have a CUresult parameter to UCT_CUDADRV_FUNC? Something like UCT_CUDADRV_FUNC(func, cu_ret)?
This would let us query the cuda return value and handle the output in a different way. Lemme know your thoughts.

Reply:
I think we can add a different macro for that when it is needed, otherwise have to define cu_ret at every usage of macro.